gnucash/report/reports/standard/income-statement.scm

```diff
@@ -494,67 +491,86 @@
-                (exp-cell (gnc:make-html-table-cell exp-table)))
-            (define (add-cells . lst) (gnc:html-table-append-row! build-table lst))
-            (cond
-             ((and two-column? standard-order?)
-              (add-cells inc-cell tra-cell exp-cell))
-
-             (two-column?
-              (add-cells exp-cell inc-cell tra-cell))
-
-             (standard-order?
-              (add-cells inc-cell)
-              (unless (null? trading-accounts) (add-cells tra-cell))
-              (add-cells exp-cell))
-
-             (else
-              (add-cells exp-cell)
-              (add-cells inc-cell)
-              (unless (null? trading-accounts) (add-cells tra-cell))))
-
-            (gnc:html-table-set-style!
-             build-table "td"
-             'attribute '("align" "left")
-             'attribute '("valign" "top"))
-            (gnc:html-document-add-object! doc build-table))
+          (define (add-revenue-table table)
+            (when label-revenue?
+              (add-subtotal-line table (G_ "Revenues") #f #f))
+            (gnc:html-table-add-account-balances table revenue-table params)
+            (when total-revenue?
+              (add-subtotal-line table (G_ "Total Revenue") #f revenue-total))
+            table)
+
+          (define (add-expense-table table)
+            (when label-expense?
+              (add-subtotal-line table (G_ "Expenses") #f #f))
+            (gnc:html-table-add-account-balances table expense-table params)
+            (when total-expense?
+              (add-subtotal-line table (G_ "Total Expenses") #f expense-total))
+            table)
+
+          (define (add-trading-table table)
+            (when label-trading?
+              (add-subtotal-line table (G_ "Trading") #f #f))
+            (gnc:html-table-add-account-balances table trading-table params)
+            (when total-trading?
+              (add-subtotal-line table (G_ "Total Trading") #f trading-total))
+            table)
+
+          (cond
+           (two-column?
+            (let* ((exp-table (add-expense-table (gnc:make-html-table)))
+                   (inc-table (add-revenue-table (gnc:make-html-table)))
+                   (tra-table (add-trading-table (gnc:make-html-table)))
+                   (inc-cell (gnc:make-html-table-cell inc-table))
+                   (tra-cell (if (null? trading-accounts)
+                                 (gnc:html-make-empty-cell)
+                                 (gnc:make-html-table-cell tra-table)))
+                   (exp-cell (gnc:make-html-table-cell exp-table)))
+              (define (add-cells . lst) (gnc:html-table-append-row! build-table lst))
+              (add-rule (if standard-order? exp-table inc-table))
+              (add-report-line
+               (if standard-order? exp-table inc-table)
+               (string-append (G_ "Net income") period-for)
+               (string-append (G_ "Net loss") period-for)
```

Comment:
Here and later: If they are headers, shouldn't the final strings be capitalized?

Reply:
Yeah, and then translated to the uncapitalized form in en_GB.po. But since @christopherlam and @Bob-IT think in English English it might be asking too much for them to remember that. "Two peoples separated by a common tongue" and all that.